Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -964,41 +988,56 @@ private extension PurchasesOrchestrator {
     func handleReceiptPost(result: Result<CustomerInfo, BackendError>,
                            subscriberAttributes: SubscriberAttribute.Dictionary,
                            completion: (@Sendable (Result<CustomerInfo, PurchasesError>) -> Void)?) {
-        operationDispatcher.dispatchOnMainThread {
-            if let customerInfo = result.value {
-                self.customerInfoManager.cache(customerInfo: customerInfo, appUserID: self.appUserID)
-            }
-
-            self.transactionPoster.markSyncedIfNeeded(subscriberAttributes: subscriberAttributes,
-                                                      error: result.error)
+        self.handlePostReceiptResult(
+            result,
+            subscriberAttributes: subscriberAttributes
+        )
 
-            if let completion = completion {
-                self.operationDispatcher.dispatchOnMainThread {
-                    completion(result.mapError { $0.asPurchasesError })
-                }
+        if let completion = completion {
+            self.operationDispatcher.dispatchOnMainThread {
+                completion(result.mapError { $0.asPurchasesError })
             }
         }
     }
 
+    func handlePostReceiptResult(_ result: Result<CustomerInfo, BackendError>,
+                                 subscriberAttributes: SubscriberAttribute.Dictionary) {
+        if let customerInfo = result.value {
+            self.customerInfoManager.cache(customerInfo: customerInfo, appUserID: self.appUserID)
+        }
+
+        self.markSyncedIfNeeded(subscriberAttributes: subscriberAttributes,
+                                error: result.error)
+    }
+
     func handlePurchasedTransaction(_ purchasedTransaction: StoreTransaction,
                                     storefront: StorefrontType?,
                                     restored: Bool) {
         let offeringID = self.getAndRemovePresentedOfferingIdentifier(for: purchasedTransaction)
+        let unsyncedAttributes = self.unsyncedAttributes
 
         self.transactionPoster.handlePurchasedTransaction(
             purchasedTransaction,
-            presentedOfferingID: offeringID,
-            storefront: storefront,
-            source: self.purchaseSource(for: purchasedTransaction.productIdentifier,
-                                        restored: restored)
-        ) { transaction, customerInfo, error, cancelled in
-            let completion = self.getAndRemovePurchaseCompletedCallback(forTransaction: purchasedTransaction)
-
-            if let customerInfo = customerInfo {
-                self.customerInfoManager.cache(customerInfo: customerInfo, appUserID: self.appUserID)
-            }
+            data: .init(
+                appUserID: self.appUserID,
+                presentedOfferingID: offeringID,
+                unsyncedAttributes: unsyncedAttributes,
```

Comment:
It seems that we were not syncing unsyncedAttributes in some of these posts. I think it shouldn't cause any issues, but it seems to be a behavior change so I just wanted to bring it up.

Reply:
That was being done inside by `TransactionPoster` because it had access to the unsynced attributes there.